### PR TITLE
benetech/ServiceNet#1056: Offer row display of records in addition to…

### PR DESCRIPTION
**Pull request by kdondziak**
… card view

### CODE REVIEW

File: src/main/webapp/app/modules/provider/all-records.tsx [hunk 1]
```diff
@@ -32,15 +39,16 @@ export interface IAllRecordsState extends IPaginationBaseState {
...
   isRecordHighlighted: boolean;
   selectedRecord: string;
   selectedLat: number;
   selectedLng: number;
 }
 
 const withLatLong = (
-  wrappedComponent: string | ComponentClass<any> | StatelessComponent<any>
-): string | React.ComponentClass<any> | React.StatelessComponent<any> => wrappedComponent;
+  wrappedComponent: string | ComponentClass<any> | FunctionComponent<any>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider/all-records.tsx:50]: What is the purpose of this change?

File: src/main/webapp/app/modules/provider/all-records.tsx [hunk 2]
```diff
@@ -258,27 +335,24 @@ export class AllRecords extends React.Component<IAllRecordsProps, IAllRecordsSta
...
-                  <FontAwesomeIcon icon="th" />
-                  <MediaQuery minDeviceWidth={768}>
-                    &nbsp;
-                    {translate('providerSite.gridView')}
-                  </MediaQuery>
-                </span>
-              </div>
-            ) : (
-              <div className="button-pill" onClick={this.toggleMapView}>
+            <div className="button-pill" onClick={this.toggleMapView}>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider/all-records.tsx]: If we are using the ButtonPill component, then I think this should be it too. For consistency.

File: src/main/webapp/app/modules/provider/all-records.tsx [hunk 3]
```diff
@@ -290,52 +364,7 @@ export class AllRecords extends React.Component<IAllRecordsProps, IAllRecordsSta
...
-              <div
-                className={`d-inline button-pill ${hasReachedMaxItems ? 'disabled' : ''}`}
-                onClick={() => this.handleLoadMore(hasReachedMaxItems)}
-              >
-                <Translate contentKey="providerSite.loadMore" />
-              </div>
-            </div>
-          </div>
-        )}
+        {isMapView ? <this.mapView /> : <this.gridView />}
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider/all-records.tsx:368]: Why these functions are in angle brackets?

File: src/main/webapp/app/modules/provider/all-records.tsx [hunk 4]
```diff
@@ -258,27 +335,24 @@ export class AllRecords extends React.Component<IAllRecordsProps, IAllRecordsSta
...
+            <div className="button-pill" onClick={this.toggleMapView}>
+              <span>
+                <FontAwesomeIcon icon={isMapView ? 'th' : 'map'} />
+                <MediaQuery minDeviceWidth={MOBILE_WIDTH_BREAKPOINT}>
+                  &nbsp;
+                  {translate(isMapView ? 'providerSite.gridView' : 'providerSite.mapView')}
+                </MediaQuery>
+              </span>
+            </div>
+            <MediaQuery minDeviceWidth={MOBILE_WIDTH_BREAKPOINT}>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider/all-records.tsx]: If it is minDeviceWidth and this button is meant for desktop I think there should be DESKTOP_WIDTH_BREAKPOINT (I had the wrong value in the previous version)?

File: src/main/webapp/app/modules/provider/all-records.tsx [hunk 5]
```diff
@@ -258,27 +335,24 @@ export class AllRecords extends React.Component<IAllRecordsProps, IAllRecordsSta
...
-                    {translate('providerSite.gridView')}
-                  </MediaQuery>
-                </span>
-              </div>
-            ) : (
-              <div className="button-pill" onClick={this.toggleMapView}>
+            <div className="button-pill" onClick={this.toggleMapView}>
+              <span>
+                <FontAwesomeIcon icon={isMapView ? 'th' : 'map'} />
+                <MediaQuery minDeviceWidth={MOBILE_WIDTH_BREAKPOINT}>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/provider/all-records.tsx]: If it is minDeviceWidth and this button is meant for desktop I think there should be DESKTOP_WIDTH_BREAKPOINT (I had the wrong value in the previous version)?

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/modules/provider/all-records.tsx (hunk 1 above)
Comment (kdondziak) [reply src/main/webapp/app/modules/provider/all-records.tsx:50]: It was marked as deprecated

File: src/main/webapp/app/modules/provider/all-records.tsx (hunk 2 above)
Comment (kdondziak) [reply src/main/webapp/app/modules/provider/all-records.tsx]: I think this should be refactored in whole service provider module. Maybe as a part of #1065. I prepared ButtonPill for that change

File: src/main/webapp/app/modules/provider/all-records.tsx (hunk 1 above)
Comment (awalkowiak) [reply src/main/webapp/app/modules/provider/all-records.tsx:50]: Ok, we have three more usages of this, please change those too (filter-activity, filter-shelter, and shelter-details)

File: src/main/webapp/app/modules/provider/all-records.tsx (hunk 3 above)
Comment (kdondziak) [reply src/main/webapp/app/modules/provider/all-records.tsx:368]: This is what I've recently learned. In my opinion it looks more like component than this.mapView().

Comment (awalkowiak) [reply src/main/webapp/app/modules/provider/all-records.tsx:368]: If eslint is fine with that, then ok

File: src/main/webapp/app/modules/provider/all-records.tsx (hunk 2 above)
Comment (awalkowiak) [reply src/main/webapp/app/modules/provider/all-records.tsx]: If you insist, then ok
